sql/group.go

```diff
@@ -178,11 +178,43 @@ func (n *groupNode) wrap(plan planNode) planNode {
 	if n == nil {
 		return plan
 	}
+	// Check to see if the requested ordering is compatible with the existing
+	// ordering.
+	existingOrdering, prefix := plan.Ordering()
+	if log.V(2) {
+		log.Infof("Group: existing=%d (%d) desired=%d", existingOrdering, prefix, n.desiredOrdering)
+	}
+
 	n.plan = plan
 	n.needGroup = true
 	return n
 }
 
+// isNotNullFilter adds as a "col IS NOT NULL" constraint to the expression if
+// the groupNode has a desired ordering on col.
```

Comment:
comment here about how this can only happen if there's a single min or max. ideally this code would imply that more naturally, though.